src/app/components/PageTitleConfirmation.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Element, EventListener } from '@openstax/types/lib.dom';
+
+export default function announcePageTitle(title: string) {
+  const el = document?.getElementById('page-title-confirmation');
+
+  el?.dispatchEvent(
+    new CustomEvent('page-title', {
+      detail: title,
+    })
+  );
+}
+
+export function PageTitleConfirmation({ className }: { className: string }) {
+  const [docTitle, setDocTitle] = React.useState('');
+  const ref = React.useRef<Element | null>(null);
+
+  React.useEffect(() => {
+    const el = ref.current;
+
+    if (!el) { return; }
+    const listener = ((e: CustomEvent) => {
+      setDocTitle(e.detail);
+    }) as EventListener;
+
+    el.addEventListener('page-title', listener);
+    return () => el.removeEventListener('page-title', listener);
+  }, []);
```

Comment:
Doesn't this need to re-trigger if the ref changes?

Reply:
The ref won't change unless there's a fresh render, but I can move the `el` definition outside the useEffect just in case.